### PR TITLE
Use "throw;" instead of "throw e;" in System.Private.Xml

### PR DESCRIPTION
**Pull request by stephentoub**
Fixes https://github.com/dotnet/corefx/issues/13106
cc: @alexperovich, @sepidehMS

### CODE REVIEW

**Comment by karelz** [comment]
cc: @krwq

File: src/System.Private.Xml/src/System/Xml/Schema/CompiledidEntityConstraint.cs [hunk 1]
```diff
@@ -56,7 +56,7 @@ public CompiledIdentityConstraint(XmlSchemaIdentityConstraint constraint, XmlNam
             catch (XmlSchemaException e)
             {
                 e.SetSource(constraint.Selector);
-                throw e;
+                throw;
```
Comment (AlexGhiondea) [inline src/System.Private.Xml/src/System/Xml/Schema/CompiledidEntityConstraint.cs]: Yes in general I am all for it! But the SetSource method seems to do something to the caught exception: 

http://source.dot.net/#System.Private.Xml/System/Xml/Schema/XmlSchemaException.cs,209

Comment (stephentoub) [inline src/System.Private.Xml/src/System/Xml/Schema/CompiledidEntityConstraint.cs]: > But the SetSource method seems to do something to the caught exception:

So does the SetValue method in my example.

Comment (AlexGhiondea) [inline src/System.Private.Xml/src/System/Xml/Schema/CompiledidEntityConstraint.cs]: Ah -- totally missed that. :)

I have to admit that this behavior is slightly surprising :) but I think it is the right behavior for throw.

LGTM!

**Pull request merged**